reactivesocket-core/src/main/java/io/reactivesocket/RequestHandler.java

```diff
@@ -81,7 +82,7 @@ public Builder withFireAndForget(final Function<Payload, Publisher<Void>> handle
             return this;
         }
 
-        public Builder withRequestChannel(final Function<Publisher<Payload> , Publisher<Payload>> handleRequestChannel) {
+        public Builder withRequestChannel(final BiFunction<Payload, Publisher<Payload> , Publisher<Payload>> handleRequestChannel) {
```

Comment:
My understanding of this flow from the protocol spec is that a client may receive either

1) initial payload with request N > 0
2) initial payload with request N == 0

The initial payload is received through the publisher as well, so this initial payload here is purely for routing purposes.  It might be worth adding a javadoc to make that clear if that is agreed.